js/src/grpc-client.ts

```diff
@@ -119,17 +119,25 @@ export class DateiLagerGrpcClient {
    * @param template     The id of the project to start from.
    */
   public async newProject(project: bigint, packPatterns: string[], template?: bigint): Promise<void> {
-    await trace(
-      "dateilager-grpc-client.new-project",
-      {
-        attributes: {
-          "dl.project": String(project),
-          "dl.pack_patterns": packPatterns,
-          "dl.template": String(template),
+    try {
+      await trace(
+        "dateilager-grpc-client.new-project",
+        {
+          attributes: {
+            "dl.project": String(project),
+            "dl.pack_patterns": packPatterns,
+            "dl.template": String(template),
+          },
         },
-      },
-      () => this._client.newProject({ id: project, packPatterns, template }, this._rpcOptions())
-    );
+        () => this._client.newProject({ id: project, packPatterns, template }, this._rpcOptions())
+      );
+    } catch (error) {
+      if (error instanceof RpcError && error.code == "ALREADY_EXISTS") {
+        throw new ProjectAlreadyExistsError(`project id ${project} already exists`);
+      } else {
+        throw error;
+      }
```

Comment:
nit: no need for an `else` block here

```suggestion
      if (error instanceof RpcError && error.code == "ALREADY_EXISTS") {
        throw new ProjectAlreadyExistsError(`project id ${project} already exists`);
      }
      throw error;
```